From f1675ec43622dfdff4f6f57c9854038da59325c4 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:47 -0400
Subject: [alsa] hda/realtek: Add eapd shutup to ALC283

Message-id: <140972823573-259-patchset-jkysela@redhat.com>
Patchwork-id: 90959
O-Subject: [RHEL 7.1 PATCH 258/442] ALSA: hda/realtek - Add eapd shutup to ALC283
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Add eapd shutup function to alc283_shutup.
It could avoid pop noise from speaker.

Signed-off-by: Kailang Yang <kailang@realtek.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 0435b3ffbaa67cded10c25e4a43404a611a7ebb5)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 35a3c95..1ba646e 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -3095,6 +3095,7 @@ static void alc283_shutup(struct hda_codec *codec)
 
  if (hp_pin_sense)
   msleep(100);
+ alc_auto_setup_eapd(codec, false);
  snd_hda_shutup_pins(codec);
  alc_write_coef_idx(codec, 0x43, 0x9614);
 }
-- 
1.7.1